src/cards/ha-media_player-card.html

```diff
@@ -219,13 +235,75 @@
       value: 1,
       reflectToAttribute: true,
     },
+
+    playbackDuration: {
+      type: Number,
+    },
+
+    playbackPosition: {
+      type: Number,
+    },
+
+    playbackVisible: {
+      type: Boolean,
+      value: false
+    },
+
+    _internal: {
+      type: Object,
+      value: {
+        position: null,  // last value of media_position
+        time: null,      // when that value was updated
+        interval: null,  // interval timer to update playbackPosition property
+      }
+    }
   },
 
   playerObjChanged: function (playerObj) {
     if (!playerObj.isOff && !playerObj.isIdle) {
       this.$.cover.style.backgroundImage = playerObj.stateObj.attributes.entity_picture ?
         'url(' + playerObj.stateObj.attributes.entity_picture + ')' : '';
     }
+
+    // clear previous interval timer
+    if (this._internal.interval) {
+      clearInterval(this._internal.interval);
+      this._internal.interval = null;
+    }
+
+    // update playback state from object
+    if (playerObj.stateObj.attributes.media_position_updated_at) {
+      var dateObj = Date.parse(playerObj.stateObj.attributes.media_position_updated_at);
+      var timeSincePositionValid = Math.abs(new Date() - dateObj) / 1000;
+      this._internal.position = playerObj.stateObj.attributes.media_position + timeSincePositionValid;
+      this.playbackDuration = playerObj.stateObj.attributes.media_duration;
+      this._internal.time = Date.now();
+    }
+    else {
+      this._internal.position = null;
+      this.playbackDuration = null;
+      this._internal.time = null;
+    }
+
+    if (playerObj.isPlaying && this._internal.position != null && this.playbackDuration) {
+      // playback is in progress, update the playback status on an interval timer
+      this._internal.interval = setInterval(this.updateProgress.bind(this), 500);
```

Comment:
Why change interval every 0.5 seconds while the progress bar has a 1 second granularity?